docs/ZMI.rst

```diff
@@ -98,3 +98,157 @@ The properties can have one of the following types:
+					<div class="col-sm-9 col-md-10">
+						<input id="title" class="form-control" type="text" name="title" 
+							value="<dtml-if title>&dtml-title;</dtml-if>" />
+					</div>
+				</div>
+				<div class="form-group">
+					<textarea id="content" data-contenttype="html" 
+						class="form-control zmi-code col-sm-12" name="data:text" wrap="off" 
+						rows="20"><dtml-var __str__></textarea>
+				</div>
+			</dtml-with>
+	
+			<div class="zmi-controls">
+				<dtml-if wl_isLocked>
+					<input class="btn btn-primary disabled" type="submit" name="submit" value="Save Changes" disabled="disabled" />
+					<span class="badge badge-warning" title="This item has been locked by WebDAV"><i class="fa fa-lock"></i></span>
+				<dtml-else>
+					<input class="btn btn-primary" type="submit" name="submit" value="Save Changes" />
+				</dtml-if>
+			</div>
+	
+		</form>
+	
+		<dtml-unless wl_isLocked>
+			<form action="manage_upload" method="post" enctype="multipart/form-data" class="zmi-upload mt-4">
+				<div class="input-group" title="Select Local File for Uploading">
+					<div class="custom-file">
+						<input type="file" name="file" class="custom-file-input" id="file-data" value="" 
+							onchange="$('.custom-file label span').html($(this).val().replace(/^.*(\\|\/|\:)/, ''));" />
+						<label class="custom-file-label" for="file-data"><span>Choose file</span></label>
+					</div>
+					<div class="input-group-append">
+						<input class="btn btn-outline-secondary" type="submit" value="Upload File" />
+					</div>
+				</div>
+			</form>
+		</dtml-unless>
+	
+	</main>
+	
+	<dtml-var manage_page_footer>
+
+More details
+------------
+
+**Textarea:** 
+A textarea element for editing template or script code uses the JS library 
+``ace`` for syntax high-lighting and line numbering. Textarea elements which 
+are declared by the CSS class ``zmi-code`` are transformed into an ace-edditor 
+field. Moreover this element has an attribute ``data-contenttype`` which is 
+needed by ace-editor to determine the fitting syntax high-lighting.
+ZPT-Example see: ``../Zope/src/Products/PageTemplates/www/ptEdit.zpt``
+
+**File upload element:** 
+The file upload element has an own form container (classfied as ``zmi-upload``). 
+All subseqeunt elements are nested as 'input-group' containing a div classified as 
+``custom-file`` nestingthe actual input element. An inline JS fired on the 
+onchange-event beautifies the file name showed after selecting it.
+ZPT-Example see: ../Zope/src/Products/PageTemplates/www/ptEdit.zpt
+
+**Hints and Warnings:** 
+Some input field show additional informations; these are added as element 
+``<small>`` directly following the referred inout field. (Both element are nested 
+by the width defining div-container). Possible text colors are declared by 
+typical bootstrap class names like ``text-warning``.
+
+**Icons:** 
+Zope4 object classes which are show in the ZMI have declared a global variable 
```

Comment:
My understanding is that these are instance variables? Is Zope Speak here different to standard python?